tests/phpunit/ContainerTest.php

```diff
@@ -109,4 +110,39 @@ public function test_it_can_build_dynamic_services(): void
 
         $this->addToAssertionCount(1);
     }
+
+    public function test_it_can_build_source_file_data_factory(): void
```

Comment:
I'm confused by that test: it can build but this ends as a fail?

Reply:
Yes, because it returns a lazy iterator, which is only trigger a failure upon use.

Reply:
what I mean is that the test is a bit confusing, at least from it's title. Maybe adding a comment where the failure occurs as to why this is expected to fail?

Reply:
True. I'll rename it.